### PR TITLE
Disable pushing

### PR DESCRIPTION
**Pull request by metalgearsloth**
Potentially controversial change:

tl;dr
Pros:
- No longer have this

https://user-images.githubusercontent.com/31366439/154447413-f6c62d2d-c766-41d5-a90b-c7347bc9a7f6.mp4

Cons:
- Easier to block people in places

So my minimum ping to USW is like 100-150ms. For the longest time there's been 4 things that make the game frustrating to play at this level of ping:
1. Airlock prediction
2. Interaction prediction
3. Pushing things
4. Footstep sound prediction (also featured in the above vid)

Airlock prediction is now done so we're left with the 3 horseman of the apocalypse.

The problem with pushing is that predicting physics is hard. Most games don't predict and just lerp server states as they come in so physics stuff looks smooth. The reason it's hard is because you can't predict external factors like other players moving which means any time they start moving or change direction the physics sim may be vastly different so you'd get teleporting everywhere from mispredicts.

At the moment we don't predict pushing and because it's server side it feels really bad to push things. When trying to predict you now also have the above problem so doing that would be difficult. Source only allows you to push small objects in multiplayer and most of them just get knocked away and can't be continually pushed.

Unless someone else comes alone and somehow makes pushing good I think it's best we just kill it because it looks and feels awful.

:cl:
- tweak: Pushing objects has been disabled. Pulling is still okay.

### CODE REVIEW

**Comment by Matz05** [comment]
Seems like this would make loading/unloading tightly-packed arrays of canisters, crates, etc. (as in a cargo shuttle if we get that) a nightmare. So easy to permanently trap yourself. It would also make Salvage potentially much more harrowing, as pulling something into an odd corner the wrong way could lead to you being wedged in and subsequently lost.

**Comment by metalgearsloth** [comment]
> as pulling something into an odd corner the wrong way could lead to you being wedged in and subsequently lost.

Pulling still disables collision.

**Comment by Matz05** [comment]
Ah, so you could pull and then walk over an object that is trapping you?

**Comment by metalgearsloth** [comment]
Yeah.

Something else we could do in future is if you push against something long enough it stops colliding with you (only putting this on lockers and crates obv).

**Comment by Matz05** [comment]
Something like a "mantle-able" property?

**Comment by mirrorcult** [comment]
im not sure i really understand the impetus here still, the video of the clown pushing stuff looks fine to me?

**Comment by Matz05** [comment]
It's the slight stutter/delay at the start of pushing and at changes in direction, I think. I personally think it's fine, I understand the limitations of light-speed and distance, and that predicting complex physics interactions is difficult.
I don't think it's bad enough to change, but others clearly have a worse experience.

**Comment by metalgearsloth** [comment]
> im not sure i really understand the impetus here still, the video of the clown pushing stuff looks fine to me?

It's significantly worse if there's another player touching the locker or pulling it at the same time but I'll get more video because it feels like garbage.

**Comment by Matz05** [comment]
Could we lock pushables/pullables so that only one person can have 'control' of them at a time? Would that help with your issue?
